src/onnx/onnx.cpp

```diff
@@ -2568,8 +2571,18 @@ struct onnx_parser
         return add_broadcastable_binary_op(cd, args[2], "add");
     }
 
-    void parse_from(std::istream& is)
+    void get_abs_path()
     {
+        path = ".";
+        if(contains(filename, '/'))
+            path = filename.substr(0, filename.find_last_of('/'));
+    }
+
+    void parse_from(std::istream& is, std::string name = "")
+    {
+        this->filename = std::move(name);
+        get_abs_path();
```

Comment:
We should probably use [`std::filesystem`](https://en.cppreference.com/w/cpp/filesystem) for this. In PR#664, I added a migraphx/filesystem.hpp header to select the filesystem available. Everything is put in the `migraphx::fs` namespace. So you should be able to compute the path with `fs::path(this->filename).parent_path()`.